packages/jest-snapshot/src/State.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import type {Path} from 'types/Config';
+
+const {SnapshotFile, forFile} = require('./SnapshotFile');
+
+class SnapshotState {
+  _counters: Map<string, number>;
+  _index: number;
+  added: number;
+  matched: number;
+  snapshot: SnapshotFile;
+  unmatched: number;
+  update: boolean;
+  updated: number;
+
+  constructor(testPath: Path, update: boolean) {
+    this._counters = new Map();
+    this._index = 0;
+    this.added = 0;
+    this.matched = 0;
+    this.snapshot = forFile(testPath);
+    this.unmatched = 0;
+    this.update = update;
+    this.updated = 0;
+  }
+
+  match(testName: string, received: any) {
+    this._counters.set(testName, (this._counters.get(testName) || 0) + 1);
+    const count = Number(this._counters.get(testName));
+
+    const key = testName + ' ' + count;
```

Comment:
`chai-jest-snapshot` was implemented without using any counters in key names, because the user has to specify each key manually (there is no runner integration there; just an assertion helper). Consequently, in https://github.com/facebook/jest/pull/1413 one of my goals was to keep the part that adds counts to the end of the testName out of the generic matcher (and just in the jest-specific one).

I could start using counts in `chai-jest-snapshot`, but it'd require a major version bump because users would need to change all their snapshot names (updating from `jest-snapshot@14.0` might already require a major version bump anyway, though).

I don't think `chai-jest-snapshot` has *that* many users, so it might not be entirely necessary to address. But in test runners where the test name has to be defined by the user, I think adding the counter can be a bit confusing and not always good UX. It makes sense with automatically generated names because it will avoid collisions, but when a user is typing the string for the snapshot name by hand, adding a 1 to the end is unexpected.